### PR TITLE
Fix divide by zero

### PR DESCRIPTION
**Pull request by Chaho12**
## Description

Fix calculating `zero/zero` when setting [StageCpuDistribution average](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/eventlistener/StageCpuDistribution.java) value and set it as 0.0 value when snapshot has no info for stage. 

e.g) Sample data for `List<StageCpuDistribution>` in [QueryStatistics](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/eventlistener/QueryStatistics.java#L128) shows that in some stages, tasks/p values is all zero so snapshot total/count is also zero.
> [{"stageId":"0","tasks":"0","p25":"0","p50":"0","p75":"0","p90":"0","p95":"0","p99":"0","min":"0","max":"0","total":"0","average":"NaN"},{"stageId":"1","tasks":"93","p25":"3","p50":"3","p75":"3","p90":"3","p95":"3","p99":"7","min":"1","max":"7","total":"274","average":"2.946236559139785"},{"stageId":"2","tasks":"1","p25":"1","p50":"1","p75":"1","p90":"1","p95":"1","p99":"1","min":"1","max":"1","total":"1","average":"1.0"},{"stageId":"3","tasks":"100","p25":"1","p50":"1","p75":"1","p90":"1","p95":"1","p99":"1","min":"0","max":"1","total":"99","average":"0.99"},{"stageId":"4","tasks":"0","p25":"0","p50":"0","p75":"0","p90":"0","p95":"0","p99":"0","min":"0","max":"0","total":"0","average":"NaN"},{"stageId":"5","tasks":"100","p25":"1","p50":"1","p75":"1","p90":"1","p95":"1","p99":"1","min":"1","max":"1","total":"100","average":"1.0"},{"stageId":"6","tasks":"0","p25":"0","p50":"0","p75":"0","p90":"0","p95":"0","p99":"0","min":"0","max":"0","total":"0","average":"NaN"},{"stageId":"7","tasks":"0","p25":"0","p50":"0","p75":"0","p90":"0","p95":"0","p99":"0","min":"0","max":"0","total":"0","average":"NaN"}]

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by sopel39** [comment]
Why `NaN` is not acceptable here?

**Comment by mosabua** [comment]
> Thanks @Chaho12 !
> 
> Why `NaN` is not acceptable here?

Mathematically NaN is correct .. but for many users they might not even clue in that this means Not a number and is a result from a division by zero. From a users perspective it feels like 0 is the correct value to display .. since the CPU load is nothing .. so zero on average...

**Comment by mosabua** [comment]
Ideally this is done for other values where this occurs as well but I think we can still improve just this one as well. Maybe update commit message though

File: core/trino-main/src/main/java/io/trino/event/QueryMonitor.java [hunk 1]
```diff
@@ -717,7 +717,7 @@ private static StageCpuDistribution computeCpuDistribution(StageInfo stageInfo)
                 (long) snapshot.getMin(),
                 (long) snapshot.getMax(),
                 (long) snapshot.getTotal(),
-                snapshot.getTotal() / snapshot.getCount());
+                snapshot.getCount() > 0 ? snapshot.getTotal() / snapshot.getCount() : 0.0);
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/event/QueryMonitor.java:720]: please use `io.trino.util.MoreMath#firstNonNaN` with static import

Comment (Chaho12) [reply core/trino-main/src/main/java/io/trino/event/QueryMonitor.java]: Fixed :)

**Comment by Chaho12** [comment]
> Ideally this is done for other values where this occurs as well but I think we can still improve just this one as well. Maybe update commit message though

I didn't look for all possible NaN's so I changed the commit message to `Return 0 instead of NaN for CPU distribution stats`

**Comment by sopel39** [comment]
thx @Chaho12

**Pull request merged**